internal/runners/config/rules.go

```diff
@@ -0,0 +1,36 @@
+package config
+
+import "github.com/ActiveState/cli/internal/constants"
+
+type event func() error
+
+var emptyEvent = func() error { return nil }
+
+type configType int
+
+const (
+	String configType = iota
+	Int
+	Bool
+)
+
+type configRule struct {
+	allowedType configType
```

Comment:
Nit: I would just drop "allowed" in the name here, it doesn't seem to add anything but it creates overhead in my parsing of the code.